src/Appwrite/Auth/Validator/MockNumber.php

```diff
@@ -47,8 +47,8 @@ public function isValid($value): bool
         }
 
         $otp = new Text(6, 6);
-        if (!$otp->isValid($value['otp'])) {
-            $this->message = 'OTP must be a valid string and exactly 6 characters.';
+        if (!$otp->isValid($value['otp']) || !\ctype_digit($value['otp'])) {
+            $this->message = 'Invalid OTP. Please make sure the OTP is a 6 digit number';
```

Comment:
Third parameter in text validator is `array $allowList = [])`, you can pass `Text::NUMBERS` to only allow numbers.